src/main/java/it/gov/pagopa/pu/debtpositions/repository/InstallmentNoPIIRepository.java

```diff
@@ -70,4 +70,9 @@ Optional<InstallmentNoPII> findAuthorizedByTransferSemanticKey(
           "   and i.nav = :nav")
   List<InstallmentNoPII> getByOrganizationIdAndNav(@Param("organizationId") Long organizationId, @Param("nav") String nav,
                                                    @Param("debtPositionOrigins") List<DebtPositionOrigin> debtPositionOrigins);
+
+  @Query(" select i" +
+    "  from InstallmentNoPII i" +
+    " where i.receiptId = :receiptId")
+  Optional<List<InstallmentNoPII>> getByReceiptId(@Param("receiptId") Long receiptId);
```

Comment:
let's keep it simple, let's get Spring generate the query for you

```suggestion
  List<InstallmentNoPII> findByReceiptId(long receiptId);
```